File: admin/src/api/erp_warehouse.ts
import { PaginatedRequest, PaginatedResponse } from '@/base/page';
import { api } from '@/utils/request';

const apis = {
  create: '/erp/erp_warehouse/create', // 新增
  update: '/erp/erp_warehouse/update', // 修改
  delete: '/erp/erp_warehouse/delete', // 删除
  get: '/erp/erp_warehouse/get', // 单条查询
  list: '/erp/erp_warehouse/list', // 列表查询
  page: '/erp/erp_warehouse/page', // 分页查询
  enable: '/erp/erp_warehouse/enable', // 启用
  disable: '/erp/erp_warehouse/disable', // 禁用
}

export interface ErpWarehouseRequest {
  id: number; // 仓库ID
  name: string; // 仓库名称
  location: string; // 仓库位置
  status: number; // 状态
  sort: number; // 排序
  storage_fee: number; // 仓储费
  handling_fee: number; // 搬运费
  manager: string; // 负责人
  remarks: string; // 备注
  department_code: string; // 部门编码
  department_id: number; // 部门ID
}

export interface ErpWarehouseResponse {
  id: number; // 仓库ID
  name: string; // 仓库名称
  location: string; // 仓库位置
  status: number; // 状态
  sort: number; // 排序
  storage_fee: number; // 仓储费
  handling_fee: number; // 搬运费
  manager: string; // 负责人
  remarks: string; // 备注
  department_code: string; // 部门编码
  department_id: number; // 部门ID
  creator: number; // 创建者ID
  create_time: string; // 创建时间
  updater: number; // 更新者ID
  update_time: string; // 更新时间
}

export interface ErpWarehouseQueryCondition extends PaginatedRequest {

}

export const createErpWarehouse = (erp_warehouse: ErpWarehouseRequest): Promise<number> => {
  return api.post<number>(apis.create, erp_warehouse);
}

export const updateErpWarehouse = (erp_warehouse: ErpWarehouseRequest): Promise<void> => {
  return api.post<void>(apis.update, erp_warehouse);
}

export const deleteErpWarehouse = (id: number): Promise<void> => {
  return api.post<void>(`${apis.delete}/${id}`);
}

export const getErpWarehouse = (id: number): Promise<ErpWarehouseResponse> => {
  return api.get<ErpWarehouseResponse>(`${apis.get}/${id}`);
}

export const listErpWarehouse = (): Promise<Array<ErpWarehouseResponse>> => {
  return api.get<Array<ErpWarehouseResponse>>(apis.list);
}

export const pageErpWarehouse = (condition: ErpWarehouseQueryCondition): Promise<PaginatedResponse<ErpWarehouseResponse>> => {
  return api.get<PaginatedResponse<ErpWarehouseResponse>>(apis.page, condition);
}

export const enableErpWarehouse = (id: number): Promise<void> => {
  return api.post<void>(`${apis.enable}/${id}`);
}

export const disableErpWarehouse = (id: number): Promise<void> => {
  return api.post<void>(`${apis.disable}/${id}`);
}

File: server/module/module-erp/erp-server/src/convert/erp_warehouse.rs
use sea_orm::{Set, NotSet};
use crate::model::erp_warehouse::{self, Model as ErpWarehouse, ActiveModel as ErpWarehouseActiveModel};
use erp_model::request::erp_warehouse::{CreateErpWarehouseRequest, UpdateErpWarehouseRequest};
use erp_model::response::erp_warehouse::ErpWarehouseResponse;

pub fn create_request_to_model(request: &CreateErpWarehouseRequest) -> ErpWarehouseActiveModel {
    ErpWarehouseActiveModel {
        name: Set(request.name.clone()),
        location: request.location.as_ref().map_or(NotSet, |location| Set(Some(location.clone()))),
        status: Set(request.status.clone()),
        sort_order: Set(request.sort_order.clone()),
        storage_fee: request.storage_fee.as_ref().map_or(NotSet, |storage_fee| Set(Some(storage_fee.clone()))),
        handling_fee: request.handling_fee.as_ref().map_or(NotSet, |handling_fee| Set(Some(handling_fee.clone()))),
        manager: request.manager.as_ref().map_or(NotSet, |manager| Set(Some(manager.clone()))),
        remarks: request.remarks.as_ref().map_or(NotSet, |remarks| Set(Some(remarks.clone()))),
        department_code: Set(request.department_code.clone()),
        department_id: Set(request.department_id.clone()),
        ..Default::default()
    }
}

pub fn update_request_to_model(request: &UpdateErpWarehouseRequest, existing: ErpWarehouse) -> ErpWarehouseActiveModel {
    let mut active_model: ErpWarehouseActiveModel = existing.into();
    if let Some(name) = &request.name {
        active_model.name = Set(name.clone());
    }
    if let Some(location) = &request.location { 
        active_model.location = Set(Some(location.clone()));
    }
    if let Some(status) = &request.status { 
        active_model.status = Set(status.clone());
    }
    if let Some(sort_order) = &request.sort_order { 
        active_model.sort_order = Set(sort_order.clone());
    }
    if let Some(storage_fee) = &request.storage_fee { 
        active_model.storage_fee = Set(Some(storage_fee.clone()));
    }
    if let Some(handling_fee) = &request.handling_fee { 
        active_model.handling_fee = Set(Some(handling_fee.clone()));
    }
    if let Some(manager) = &request.manager { 
        active_model.manager = Set(Some(manager.clone()));
    }
    if let Some(remarks) = &request.remarks { 
        active_model.remarks = Set(Some(remarks.clone()));
    }
    if let Some(department_code) = &request.department_code { 
        active_model.department_code = Set(department_code.clone());
    }
    if let Some(department_id) = &request.department_id { 
        active_model.department_id = Set(department_id.clone());
    }
    active_model
}

pub fn model_to_response(model: ErpWarehouse) -> ErpWarehouseResponse {
    ErpWarehouseResponse { 
        id: model.id,
        name: model.name,
        location: model.location,
        status: model.status,
        sort_order: model.sort_order,
        storage_fee: model.storage_fee,
        handling_fee: model.handling_fee,
        manager: model.manager,
        remarks: model.remarks,
        department_code: model.department_code,
        department_id: model.department_id,
        creator: model.creator,
        create_time: model.create_time,
        updater: model.updater,
        update_time: model.update_time,
    }
}